docs/source/plugins.rst

```diff
@@ -0,0 +1,200 @@
+.. _plugins:
+
+==============
+MLflow Plugins
+==============
+
+As a framework-agnostic tool for machine learning, the MLflow Python API provides developer APIs for
+writing plugins that integrate with different ML frameworks and backends.
+
+Plugins provide a powerful mechanism for customizing the behavior of the MLflow
+Python client and integrating third-party tools, allowing you to:
+
+- Integrate with third-party storage solutions for experiment data, artifacts, and models
+- Integrate with third-party authentication providers, e.g. read HTTP authentication credentials
+  from a special file
+- Use the MLflow client to communicate with other REST APIs, e.g. your organization's existing
+  experiment-tracking APIs
+- Automatically capture additional metadata as run tags, e.g. the git repository associated with a run
+
+The MLflow Python API currently supports several types of plugins:
```

Comment:
Remove **currently**. Documentation describes the state of a release. Currently is implied.